content/blog/tune-parallel/index.Rmd

```diff
@@ -0,0 +1,181 @@
+---
+output: hugodown::hugo_document
+
+slug: tune-parallel
+title: Parallel processing with tune 
+date: 2020-11-18
+author: Max Kuhn
+description: >
+    With version 0.1.2 of tune, there are more options for parallel processing.  
+
+photo:
+  url: https://unsplash.com/photos/mJ35U595uhA
+  author: Joss Woodhead
+
+# one of: "deep-dive", "learn", "package", "programming", or "other"
+categories: [package] 
+tags: [tidymodels,tune, parallelism]
+---
+
+```{r startup, include = FALSE}
+library(ggplot2)
+thm <- theme_bw() + 
+  theme(
+    panel.background = element_rect(fill = "transparent", colour = NA), 
+    plot.background = element_rect(fill = "transparent", colour = NA),
+    legend.background = element_rect(fill = "transparent", colour = NA),
+    legend.key = element_rect(fill = "transparent", colour = NA)
+  )
+theme_set(thm)
+```
+
+<!--
+TODO:
+* [ ] Pick category and tags (see existing with `post_tags()`)
+* [ ] Find photo & update yaml metadata
+* [ ] Create `thumbnail-sq.jpg`; height and width should be equal
+* [ ] Create `thumbnail-wd.jpg`; width should be >5x height
+* [ ] `hugodown::use_tidy_thumbnail()`
+* [ ] Add intro sentence
+* [ ] `use_tidy_thanks()`
+-->
+This is the second post related to version 0.1.2 of the tune package. The [first post]() discussed various new features. This post is an excerpt from an upcoming chapter in [_Tidy Modeling with R_](https://www.tmwr.org/) and is focused on parallel processing. 
+
+Previously, the tune package allowed for parallel processing of calculations in a few different places: 
+
+* Simple model resampling via `resample_fit()`
+
+* Model tuning via `tune_grid()`
+
+* During Bayesian optimization (`tune_bayes()`)
+
+In the new version of tune, there are more options related to how parallelism occurs. It's a little complicated and we'll start by describing the most basic method. 
+
+## Parallelizing the resampling loop
+
+For illustration, let's suppose that we are tuning a set of model parameters (e.g. not recipe parameters). In tidymodels, we always use [out-of-sample predictions to measure performance](https://www.tmwr.org/resampling.html). With grid search, pseudo-code that illustrations the computations are: 
+
+```{r eval = FALSE}
+for (rs in resamples) {
+   # Create analysis and assessment sets
+   # Preprocess data (e.g. formula or recipe)
+   for (mod in configurations) {
+      # Fit model {mod} to the {rs} analysis set
+      # Predict the {rs} assessment set
```

Comment:
I think {rs} is a little unclear, maybe just {resample} or at least something a little longer?